tensordict/nn/common.py

```diff
@@ -39,10 +39,25 @@ class FunctionalModuleWithBuffers:
 def _check_all_str(sequence_of_str: Sequence[str]) -> None:
     if isinstance(sequence_of_str, str):
         raise RuntimeError(
-            f"Expected a sequence of strings but got a string: {sequence_of_str}"
+            f"Expected a sequence of strings or tuples of strings, but got a string: {sequence_of_str}"
         )
-    if any(not isinstance(key, str) for key in sequence_of_str):
-        raise TypeError(f"Expected a sequence of strings but got: {sequence_of_str}")
+    for key in sequence_of_str:
+        if isinstance(key, str):
+            continue
+        elif isinstance(key, tuple):
+            if not len(key):
+                raise RuntimeError("Empty tuples are not permitted.")
+            for sub_key in key:
+                if isinstance(sub_key, str):
+                    continue
+                else:
+                    RuntimeError(
+                        f"Expected a sequence of strings or tuples of strings, but got a string: {sequence_of_str}"
```

Comment:
I think we can lose `a string` from this error message. It makes sense on line 42, but here the `sub_key` could be any type

Reply:
You're right
I addresses the comments, can you have a look? It's ready on my side